scripts/lib/CIME/case.py

```diff
@@ -443,13 +443,11 @@ def _set_compset_and_pesfile(self, compset_name, files, user_compset=False, pesf
                     user_mods_dir     = files.get_value("USER_MODS_DIR"     , {"component":component}, resolved=False)
                     self.set_lookup_value("COMPSETS_SPEC_FILE" ,
                                    files.get_value("COMPSETS_SPEC_FILE", {"component":component}, resolved=False))
+                    self._principle_component = component.upper()
```

Comment:
I would like this PR to go in without the support for user-defined compsets - which I agree needs to go into a new PR that could be integrated with #608.